components/RunsPage/Header/Header.vue

```diff
@@ -0,0 +1,58 @@
+<script lang="ts" setup>
+import type { LeaderboardViewModel } from '~/lib/api/data-contracts'
+import { useGetCategoriesForLeaderboard } from '~/composables/api'
+import ButtonLink from '~/components/elements/buttons/ButtonLink/ButtonLink.vue'
+
+interface HeaderProps {
+  leaderboard?: LeaderboardViewModel
```

Comment:
I don't think this should be null. Semantically, this component needs the leaderboard to exist, otherwise it doesn't really work. Instead, consider checking in the parent component for an error when fetching the board and displaying that error instead of this component.